File: SwrveSDKTest/SDKTests/Universal/SwrveTestABTestDetails.m
#import <XCTest/XCTest.h>
#import "SwrveTestHelper.h"

@interface SwrveTestABTestDetails : XCTestCase

@end

@implementation SwrveTestABTestDetails

- (void) testABTestDetails {
    SwrveConfig* config = [[SwrveConfig alloc] init];
    config.abTestDetailsEnabled = YES;
    
    Swrve *swrve = [SwrveTestHelper initializeSwrveWithCampaignsFile:@"abTestDetails" andConfig:config];
    
    // Assert it has the loaded AB Test Details
    NSArray* abTestDetails = [swrve.resourceManager abTestDetails];
    XCTAssertEqual(2, [abTestDetails count]);
    
    for (SwrveABTestDetails *details in abTestDetails) {
        if ([details.name isEqualToString:@"AB test Name 1"]) {
            XCTAssertEqualObjects(@"12", details.id);
            XCTAssertEqual(1, details.caseIndex);
        } else if ([details.name isEqualToString:@"AB test Name 2"]) {
            XCTAssertEqualObjects(@"13", details.id);
            XCTAssertEqual(4, details.caseIndex);
        } else {
            XCTFail(@"Unexpected Name");
        }
    }
}

@end

File: SwrveSDKTest/SDKTests/Universal/SwrveTestABTestDetails.swift
import XCTest

class SwrveTestABTestDetails: XCTestCase {

    func testABTestDetails() {
        let config = SwrveConfig()
        config.abTestDetailsEnabled = true

        guard let swrve = SwrveTestHelper.initializeSwrve(withCampaignsFile: "abTestDetails", andConfig: config) else {
            XCTFail("Failed to initialize Swrve")
            return
        }

        // Assert it has the loaded AB Test Details
        guard let abTestDetails = swrve.resourceManager?.abTestDetails() else {
            XCTFail("AB Test Details are nil")
            return
        }

        XCTAssertEqual(abTestDetails.count, 2)

        for details in abTestDetails {
            if details.name == "AB test Name 1" {
                XCTAssertEqual(details.id, "12")
                XCTAssertEqual(details.caseIndex, 1)
            } else if details.name == "AB test Name 2" {
                XCTAssertEqual(details.id, "13")
                XCTAssertEqual(details.caseIndex, 4)
            } else {
                XCTFail("Unexpected Name")
            }
        }
    }
}
